roles/edpm_ovn/tasks/configure.yml

```diff
@@ -14,6 +14,33 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+- name: Apply ovsdb configuration from configmaps
+  block:
+  - name: discover configmap files in {{ edpm_ovn_config_src }}
+    ansible.builtin.find:
+      paths: "{{ edpm_ovn_config_src }}"
+      file_type: file
+      recurse: yes
+    register: edpm_ovn_configmaps
+    changed_when: false
+    check_mode: false
+    delegate_to: localhost
+
+  - name: read configmap file contents
+    ansible.builtin.slurp:
+      src: "{{ item.path }}"
+    loop: "{{ edpm_ovn_configmaps.files }}"
+    register: ovn_configmap_contents
+    delegate_to: localhost
+    when: edpm_ovn_configmaps.matched > 0
+
+  - name: apply configuration to Open_vSwitch:external_ids
+    ansible.builtin.shell: |
```

Comment:
shouldn't this be done after line 70 else defaults would override the settings from config map?

Reply:
I wasn't sure what's better. It may be said that it's wise to allow to override the values passed from CP layer in EDP NodeTemplate if needed. Otherwise you can't e.g. tweak timeouts for a node or the set of EDP nodes without affecting CP nodes / other EDP nodes.

Reply:
Yeah very good point and we should figure this out which one should be higher priority if config is allowed via both options and see what should be set by OVN Controller CRs and what with dataplane CRs. Likely node specific settings(timeouts etc) should be not set with Control Plane CRs? And things like "ovn-remote" to be set with data generated by Control Plane CRs.

Reply:
I actually wasn't sure if timeout proxy from CP to EDP is the right thing to do (asked about this before). The remote and tunnel type are definitely to be in sync, as for the rest - not sure. But maybe as "defaults" open for override - yes?

Reply:
yes agree ovn-remote and ovn-tunnel-type to be in sync and set with OVNController CR. For other node specific bits feels like it would be good to handle from dataplane CR side as per role/node. But let's see what others say.
I  not quite understood the point re. "defaults" and override as what defaults and override you mean.

Reply:
@karelyatin re defaults: we may look at "timeout" values set in OVNController as "defaults" that are passed to EDP through ConfigMap. Then the admin that defines nodeTempate may override the timeout values using one of `edpm_` variables for the OVN playbook.

Since the order of the actions in the current PR version is:
1. apply configMap,
2. apply external_ids from `edpm_*` variable,

this "override" should work.

If you don't need to override the "default" values from configMap, you don't specify them in nodeTemplate.

Reply:
@booxter Thanks now got it what you meant. Actually as pointed initially with current implementation defaults from edpm will always override what's set with Configmap so current version needs to be fixed.

Reply:
Great catch. I fixed it. Also included molecule to test this.